source/Ship.cpp

```diff
@@ -3966,6 +3993,34 @@ double Ship::BestFuel(const string &type, const string &subtype, double defaultF
 			if(jumpRange >= jumpDistance)
 			{
 				double fuel = it.first->Get("jump fuel");
+				// If the jump drive or hyperdrive has an attached
+				// (mass/distance) (reference/exponent), calculate
+				// the fuel about to be used for this jump.
+				double mass = Mass();
+				double driveMassExp = 0;
+				double driveMassRef = 400;
+				double driveDistExp = 0;
+				double driveDistRef = 100;
+
+				if((it.first->Get("drive mass exponent") || it.first->Get("drive distance exponent")) && mass > 0.)
+				{
+					driveMassRef = max(it.first->Get("drive mass reference"), .01);
+					driveMassExp = max(it.first->Get("drive mass exponent"), 0.);
+					fuel *= pow((mass / driveMassRef), driveMassExp);
+				}
+				if((it.first->Get("drive mass reference") || it.first->Get("drive distance reference")) && jumpDistance > 0.)
+				{
+					driveDistRef = max(it.first->Get("drive distance reference"), .01);
+					driveDistExp = max(it.first->Get("drive distance exponent"), 0.);
+					fuel *= pow((jumpDistance / driveDistRef), driveDistExp);
+				}
+
+				// if a "startup" fuel is provided, add that to the above formula.
+				// It's a constant fuel consumption regardless of mass or distance of jump.
+				if(it.first->Get("jump startup fuel") > 0)
+					fuel += it.first->Get("jump startup fuel");
```

Comment:
You're using the exact same logic twice, try to place it in a lambda (do contact me on the discord if you need help for that)
I already applied the suggestions from above on it.

The only information you need here is fuel so that's what the lambda should be returning.

It should be like this, and declared before you used it, on line 3949:
```cpp
auto calculateFuelCost = [this, jumpDistance](const Outfit &outfit) -> double
{
	double fuel = outfit.Get("jump fuel");
	// If the jump drive or hyperdrive has an attached
	// (mass/distance) (reference/exponent), calculate
	// the fuel about to be used for this jump.
	double mass = Mass();
	double driveMassExp = 0;
	double driveMassRef = 400;
	double driveDistExp = 0;
	double driveDistRef = 100;

	if((outfit.Get("drive mass exponent") || outfit.Get("drive distance exponent")))
	{
		driveMassRef = max(outfit.Get("drive mass reference"), .01);
		driveMassExp = max(outfit.Get("drive mass exponent"), 0.);
		fuel *= pow((mass / driveMassRef), driveMassExp);
	}
	if((outfit.Get("drive mass reference") || outfit.Get("drive distance reference")) && jumpDistance > 0.)
	{
		driveDistRef = max(outfit.Get("drive distance reference"), .01);
		driveDistExp = max(outfit.Get("drive distance exponent"), 0.);
		fuel *= pow((jumpDistance / driveDistRef), driveDistExp);
	}

	// if a "startup" fuel is provided, add that to the above formula.
	// It's a constant fuel consumption regardless of mass or distance of jump.
	fuel += outfit.Get("jump startup fuel");

	return fuel;
};
```
then you can call it here, using this: `double fuel = calculateFuelCost(*it.first)`